cpu/cortexm_common/include/irq_arch.h

```diff
@@ -18,41 +18,59 @@
  * @}
  */
 
+
+#ifndef _IRQ_ARCH_H
+#define _IRQ_ARCH_H
+
 #include <stdint.h>
-#include "irq.h"
-#include "cpu.h"
+#include "cpu_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /**
  * @brief Disable all maskable interrupts
  */
-unsigned int irq_disable(void)
+static inline __attribute__((always_inline)) unsigned int irq_disable(void)
 {
     uint32_t mask = __get_PRIMASK();
+
     __disable_irq();
     return mask;
 }
 
 /**
  * @brief Enable all maskable interrupts
  */
-__attribute__((used)) unsigned int irq_enable(void)
+static inline __attribute__((always_inline)) __attribute__((used)) unsigned int
+irq_enable(void)
 {
+    unsigned result = __get_PRIMASK();
     __enable_irq();
-    return __get_PRIMASK();
+    return result;
```

Comment:
What's the actual difference here regarding the original code? By inlining the function it is the "extra" variable required? By looking at the sizes it's obviously not affecting, just a question out of curiosity.

Reply:
The return value here is never used in the whole code base of RIOT. There was a lengthy discussion about dropping it altogether that somehow never concluded to just dropping it. Maybe it is time again to revisit this PR and check if we now can just get rid of that.

But back to the size question: I think for the assembly it is just two instructions in different order now, as `__STATIC_FORCEINLINE __enable_irq(void)` leaves the register used to pass the return value untouched (according to the clobbers in the inline assembly). So the variable should be only present in C, but not in the machine code.

Reply:
I can't find the PR anymore :-/